backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/grssignum/interceptor/ExternalSystem_Config_GRSSignum.java

```diff
@@ -91,4 +92,24 @@ public void updateIsAutoFlag(final I_ExternalSystem_Config_GRSSignum grsConfig)
 		}
 	}
 
+	@ModelChange(timings = { ModelValidator.TYPE_AFTER_CHANGE },
+			ifColumnsChanged = {
+					I_ExternalSystem_Config_GRSSignum.COLUMNNAME_IsCreateBPartnerFolders,
+					I_ExternalSystem_Config_GRSSignum.COLUMNNAME_BasePathForExportDirectories,
+					I_ExternalSystem_Config_GRSSignum.COLUMNNAME_BPartnerExportDirectories })
+	public void checkIsCreateBPartnerFoldersFlag(final I_ExternalSystem_Config_GRSSignum grsConfig)
+	{
+		if (!grsConfig.isCreateBPartnerFolders())
+		{
+			return;
+		}
+
+		if (Check.isBlank(grsConfig.getBPartnerExportDirectories())
+				|| Check.isBlank(grsConfig.getBasePathForExportDirectories()))
+		{
+			throw new AdempiereException("BPartnerExportDirectories and BasePathForExportDirectories must be set!")
```

Comment:
are we expecting this error be shown to the user?
probably not, since there is a MandatoryLogic in AD_Column, right? then the message is fine like this

Reply:
yes, right. this one will not get displayed